### ISSUE TITLE
String Interpolation in the configuration parser is broken for newline characters alike (WAS  mutate->gsub doesn't work on json extracted field with backslash)

### ISSUE THREAD

**Issue by nellicus** [elastic/logstash#3238]
OK
gsub replacing "\n" works on plain message field

```
abonuccelli@w530 /opt/elk/TEST/logstash-1.5.0-rc3 $ ./bin/logstash -e 'input{stdin{}}filter{mutate{gsub=>["message","\\n","newline"]}}output{stdout{codec=>rubydebug}}'
Logstash startup completed
\n
{
       "message" => "newline",
      "@version" => "1",
    "@timestamp" => "2015-05-13T10:09:34.542Z",
          "host" => "w530"
}
```

PROBLEM
gsub replacing "\n" doesn't work on json field

```
abonuccelli@w530 /opt/elk/TEST/logstash-1.5.0-rc3 $ ./bin/logstash -e 'input{stdin{}}filter{json{source=>message}mutate{gsub=>["jsonfield","\\n","newline"]}}output{stdout{codec=>rubydebug}}' 
Logstash startup completed
{"jsonfield":"\n"}
{
       "message" => "{\"jsonfield\":\"\\n\"}",
      "@version" => "1",
    "@timestamp" => "2015-05-13T10:10:23.311Z",
          "host" => "w530",
     "jsonfield" => "\n"
}
```

note this seems something specific with replacing slashes in json - replacement in json field with no \ works
```
/opt/elk/TEST/logstash-1.5.0-rc3 $ ./bin/logstash -e 'input{stdin{}}filter{json{source=>message}mutate{gsub=>["jsonfield","apples","oranges"]}}output{stdout{codec=>rubydebug}}' 
Logstash startup completed
{"jsonfield":"apples"}
{
       "message" => "{\"jsonfield\":\"apples\"}",
      "@version" => "1",
    "@timestamp" => "2015-05-13T10:08:23.933Z",
          "host" => "w530",
     "jsonfield" => "oranges"
}
```

**Comment by ph** [comment]
Same behavior in 1.4.2, so it's not related to the *jrjackson*
```
~/e/s/logstash-1.4.2 ❯❯❯ ./bin/logstash -e 'input{stdin{}}filter{json{source=>message}mutate{gsub=>["jsonfield","\\n","newline"]}}output{stdout{codec=>rubydebug}}'
Using milestone 2 filter plugin 'json'. This plugin should be stable, but if you see strange behavior, please let us know! For more information on plugin milestones, see http://logstash.net/docs/1.4.2/plugin-milestones {:level=>:warn}
{"jsonfield":"\n"}
{
       "message" => "{\"jsonfield\":\"\\n\"}",
      "@version" => "1",
    "@timestamp" => "2015-05-13T13:25:50.303Z",
          "host" => "sashimi",
     "jsonfield" => "\n"
}
```

**Comment by purbon** [comment]
@ph event accessors ?

**Comment by jordansissel** [comment]
For Logstash 1.5.0, we've moved all plugins to individual repositories, so I have moved this issue to https://github.com/logstash-plugins/logstash-filter-mutate/issues/32. Let's continue the discussion there! :)

**Comment by purbon** [comment]
@jordansissel as in #3259, is was really not an error of gsub, but an error in the configuration utf8 transformations, so the issue belongs in here.

**Comment by purbon** [comment]
Closed in favor of #3265
